Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from fontTools import ttLib
+from fontTools.pens.hashPointPen import HashPointPen
+from fontTools.ttLib.tables._g_l_y_f import (
+    OVERLAP_COMPOUND,
+    ROUND_XY_TO_GRID,
+    USE_MY_METRICS,
+)
+from ufo2ft.constants import (
+    TRUETYPE_INSTRUCTIONS_KEY,
+    TRUETYPE_ROUND_KEY,
+    TRUETYPE_METRICS_KEY,
+    TRUETYPE_OVERLAP_KEY,
+    OBJECT_LIBS_KEY,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class InstructionCompiler(object):
+    def __init__(self, ufo, ttf):
+        self.ufo = ufo
+        self.font = ttf
+
+    def _compile_program(self, key, table_tag):
+        assert table_tag in ("prep", "fpgm")
+        ttdata = self.ufo.lib.get(TRUETYPE_INSTRUCTIONS_KEY, None)
+        if ttdata:
+            formatVersion = ttdata.get("formatVersion", None)
+            if int(formatVersion) != 1:
+                logger.error(
```

Comment:
maybe consider raising a NotImplementedError instead of a logging.error which could easily be ignored, unless ignoring is desirable

Reply:
Ok